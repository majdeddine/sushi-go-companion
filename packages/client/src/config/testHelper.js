import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { Theme } from "../resources/styles/theme";

const renderWithTheme = component =>
  renderer.create(<ThemeProvider theme={Theme}>{component}</ThemeProvider>);

export { renderWithTheme };
