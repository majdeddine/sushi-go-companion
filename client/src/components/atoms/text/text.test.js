import React from 'react'
import Text from './text'
import { renderWithTheme } from '../../../config/testHelper'

jest.mock('resources/styles/theme')
jest.mock('utils/style/helpers')

describe('<Text/>', () => {
  it('Should render a Text element with default props', () => {
    const component = renderWithTheme(<Text>Text Paragraph</Text>).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('Should render a Text element with given props', () => {
    const props = {
      color: 'light',
      weight: 'bold',
    }
    const component = renderWithTheme(<Text {...props}>Text Paragraph</Text>).toJSON()
    expect(component).toMatchSnapshot()
  })
})
