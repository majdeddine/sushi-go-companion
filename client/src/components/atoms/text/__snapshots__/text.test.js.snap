// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> Should render a Text element with default props 1`] = `
<p
  className="text__Text-sc-109mrig-0 dgcKge"
  color="dark"
  size="medium"
>
  Text Paragraph
</p>
`;

exports[`<Text/> Should render a Text element with given props 1`] = `
<p
  className="text__Text-sc-109mrig-0 cceaSe"
  color="light"
  size="medium"
>
  Text Paragraph
</p>
`;
